### ISSUE TITLE
AD0001	Analyzer 'FakeItEasy.Analyzer.UnusedReturnValueAnalyzer' threw an exception

### ISSUE THREAD

**Issue by Mertsch** [FakeItEasy/FakeItEasy#731]
I was trying out https://www.nuget.org/packages/FakeItEasy.Analyzer/
and got a bunch of warnings, all in the same style.
The `SyntaxNode` exists, but isn't called in any way by FakeItEasy (`A.`...) code.

Don't know if you guys can make something of this, if not then close as I wont be spending time analyzing this further.

```
Severity	Code	Description	Project	File	Line	Source	Suppression State	Detail Description
Warning	AD0001	Analyzer 'FakeItEasy.Analyzer.UnusedReturnValueAnalyzer' threw an exception of type 'System.ArgumentNullException' with message 'Value cannot be null.
Parameter name: syntax'.	<Project>		1	IntelliSense	Active	Analyzer 'FakeItEasy.Analyzer.UnusedReturnValueAnalyzer' threw the following exception:
'Exception occurred with following context:
Compilation: <Project>
SyntaxTree: C:\Source\<Project>\AnalyzerTests.cs
SyntaxNode: CreateId() [InvocationExpressionSyntax]@[4016..4026) (118,46)-(118,56)

System.ArgumentNullException: Value cannot be null.
Parameter name: syntax
   at Microsoft.CodeAnalysis.CSharp.CSharpSemanticModel.CheckSyntaxNode(CSharpSyntaxNode syntax)
   at Microsoft.CodeAnalysis.CSharp.CSharpSemanticModel.GetSymbolInfo(ExpressionSyntax expression, CancellationToken cancellationToken)
   at Microsoft.CodeAnalysis.CSharp.CSharpExtensions.GetSymbolInfo(SemanticModel semanticModel, ExpressionSyntax expression, CancellationToken cancellationToken)
   at FakeItEasy.Analyzer.UnusedReturnValueAnalyzer.GetCalledMethodSymbol(InvocationExpressionSyntax call, SyntaxNodeAnalysisContext context)
   at FakeItEasy.Analyzer.UnusedReturnValueAnalyzer.AnalyzeCall(SyntaxNodeAnalysisContext context)
   at Microsoft.CodeAnalysis.Diagnostics.AnalyzerExecutor.<>c__DisplayClass42_1`1.<ExecuteSyntaxNodeAction>b__1()
   at Microsoft.CodeAnalysis.Diagnostics.AnalyzerExecutor.ExecuteAndCatchIfThrows_NoLock(DiagnosticAnalyzer analyzer, Action analyze, Nullable`1 info)
-----
'.
```

**Comment by adamralph** [comment]
@Mertsch thank you for raising this.

Can you please post the line of code which causing this, i.e. `AnalyzerTests.cs (118,46)-(118,56)` (I think).

Without that, it's difficult to know how to go about reproducing this.

**Comment by Mertsch** [comment]
```
        [MustUseReturnValue]
        protected static int CreateId()
        {
            return lastId++;
        }
[...]
118:            MyObject obj = new MyObject(ConstId, CreateId(), true);
[...]
```

**Comment by adamralph** [comment]
@Mertsch thanks, but I'm a little confused.

Where is `MustReturnValue` defined? There was pre-release version of FakeItEasy 2.0 which defined and used to mark method for the analyzer but it was internal, and has since been removed.

Are you working on the FakeItEasy codebase or are using you using FakeItEasy in one of your own test projects?

**Comment by Mertsch** [comment]
@adamralph Oh, sorry for the confusion. I am using FakeItEasy in my own codebase as nuget package. Additionally I do use the R# Annotations package to annotate my codebase (including test code).

So the attribute is not related to FakeItEasy at all.
Using https://www.nuget.org/packages/JetBrains.Annotations/ 10.1.4

**Comment by adamralph** [comment]
@Mertsch OK, thanks. I've managed to reproduce it now. The use of JetBrains.Annotations appears to be irrelevant:

1. Create a class library
2. `Install-Package FakeItEasy.Analyzer`
3. Add this code:
```C#
namespace Test
{
    public class Foo
    {
        public int Bar()
        {
            return 0;
        }

        public void Baz()
        {
            Bar();
        }
    }
}
```

**Comment by blairconrad** [comment]
This issue has been fixed in [release 2.1.0](https://github.com/FakeItEasy/FakeItEasy/releases/tag/2.1.0).

Thanks for your help, @Mertsch. Look for your name in the release notes! :trophy:
